test/UniswapV2Router01.spec.ts

```diff
@@ -16,546 +16,569 @@ const overrides = {
+        const WETHPartnerAmount = expandTo18Decimals(10)
+        const ETHAmount = expandTo18Decimals(5)
+        await WETHPartner.transfer(WETHPair.address, WETHPartnerAmount)
+        await WETH.deposit({ value: ETHAmount })
+        await WETH.transfer(WETHPair.address, ETHAmount)
+        await WETHPair.mint(wallet.address, overrides)
+
+        const swapAmount = expandTo18Decimals(1)
+        const expectedOutputAmount = bigNumberify('1662497915624478906')
+        const WETHPairToken0 = await WETHPair.token0()
+        await expect(
+          router.swapExactETHForTokens(0, [WETH.address, WETHPartner.address], wallet.address, MaxUint256, {
+            ...overrides,
+            value: swapAmount
+          })
+        )
+          .to.emit(WETH, 'Transfer')
+          .withArgs(router.address, WETHPair.address, swapAmount)
+          .to.emit(WETHPartner, 'Transfer')
+          .withArgs(WETHPair.address, wallet.address, expectedOutputAmount)
+          .to.emit(WETHPair, 'Sync')
+          .withArgs(
+            WETHPairToken0 === WETHPartner.address
+              ? WETHPartnerAmount.sub(expectedOutputAmount)
+              : ETHAmount.add(swapAmount),
+            WETHPairToken0 === WETHPartner.address
+              ? ETHAmount.add(swapAmount)
+              : WETHPartnerAmount.sub(expectedOutputAmount)
+          )
+          .to.emit(WETHPair, 'Swap')
+          .withArgs(
+            router.address,
+            WETHPairToken0 === WETHPartner.address ? 0 : swapAmount,
+            WETHPairToken0 === WETHPartner.address ? swapAmount : 0,
+            WETHPairToken0 === WETHPartner.address ? expectedOutputAmount : 0,
+            WETHPairToken0 === WETHPartner.address ? 0 : expectedOutputAmount,
+            wallet.address
+          )
+      })
+
+      it('swapExactETHForTokens:gas', async () => {
+        const WETHPartnerAmount = expandTo18Decimals(10)
+        const ETHAmount = expandTo18Decimals(5)
+        await WETHPartner.transfer(WETHPair.address, WETHPartnerAmount)
+        await WETH.deposit({ value: ETHAmount })
+        await WETH.transfer(WETHPair.address, ETHAmount)
+        await WETHPair.mint(wallet.address, overrides)
+
+        // ensure that setting price{0,1}CumulativeLast for the first time doesn't affect our gas math
+        await mineBlock(provider, (await provider.getBlock('latest')).timestamp + 1)
+        await pair.sync(overrides)
+
+        const swapAmount = expandTo18Decimals(1)
+        await mineBlock(provider, (await provider.getBlock('latest')).timestamp + 1)
+        const tx = await router.swapExactETHForTokens(
+          0,
+          [WETH.address, WETHPartner.address],
+          wallet.address,
+          MaxUint256,
+          {
+            ...overrides,
+            value: swapAmount
+          }
+        )
+        const receipt = await tx.wait()
+        expect(receipt.gasUsed).to.eq(routerVersion === RouterVersion.UniswapV2Router01 ? 138770 : 138770)
```

Comment:
same here

Reply:
yes, you'll see it's reworked in a later PR